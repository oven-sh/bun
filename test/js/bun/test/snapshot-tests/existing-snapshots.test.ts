test("it will work with an existing snapshot file made with bun", () => {
  expect({ a: { b: { c: false } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(Boolean) } } });
  expect({ a: { b: { c: "string" } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(String) } } });
  expect({ a: { b: { c: 4 } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(Number) } } });
  expect({ a: { b: { c: 2n } }, c: 2, jkfje: 99238 }).toMatchSnapshot({ a: { b: { c: expect.any(BigInt) } } });
  expect({ a: new Date() }).toMatchSnapshot({ a: expect.any(Date) });
  expect({ j: 2, a: "any", b: "any2" }).toMatchSnapshot({ j: expect.any(Number), a: "any", b: expect.any(String) });
  expect({ j: /regex/, a: "any", b: "any2" }).toMatchSnapshot({
    j: expect.any(RegExp),
    a: "any",
    b: expect.any(String),
  });
});
