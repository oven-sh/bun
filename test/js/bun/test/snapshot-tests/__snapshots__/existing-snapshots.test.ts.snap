// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`it will work with an existing snapshot file made with bun 1`] = `
{
  "a": {
    "b": {
      "c": Any<Boolean>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will work with an existing snapshot file made with bun 2`] = `
{
  "a": {
    "b": {
      "c": Any<String>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will work with an existing snapshot file made with bun 3`] = `
{
  "a": {
    "b": {
      "c": Any<Number>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will work with an existing snapshot file made with bun 4`] = `
{
  "a": {
    "b": {
      "c": Any<BigInt>,
    },
  },
  "c": 2,
  "jkfje": 99238,
}
`;

exports[`it will work with an existing snapshot file made with bun 5`] = `
{
  "a": Any<Date>,
}
`;

exports[`it will work with an existing snapshot file made with bun 6`] = `
{
  "a": "any",
  "b": Any<String>,
  "j": Any<Number>,
}
`;

exports[`it will work with an existing snapshot file made with bun 7`] = `
{
  "a": "any",
  "b": Any<String>,
  "j": Any<RegExp>,
}
`;
